### ISSUE TITLE
Seeing "No active lock, please report this as a bug."

### ISSUE THREAD

**Issue by m-mercuri** [lukas-krecan/ShedLock#1493]
**Describe the bug**
Sporadically, I'm getting this message "No active lock, please report this as a bug.", both locally and in production. Let me know what other information I can provide.

1. Which version do you use: 
> 4.44.0
2. Which Lock Provider: 
> JdbcTemplate: `JdbcTemplateLockProvider.Configuration.builder().withJdbcTemplate(JdbcTemplate(dataSource)).usingDbTime().build(),`
3. ShedLock configuration: 
> lockAtMostFor: 5 min
> lockAtLeastFor: 3 seconds
4. ShedLock logs:
> Not much on that thread, but for this example I noted that the locked process took 10 seconds to complete:
> ```
>2023-07-21 00:03:58.359  WARN 1 --- [   scheduling-9] n.j.s.core.DefaultLockingTaskExecutor    : No active lock, please >report this as a bug.
>```

**Comment by lukas-krecan** [comment]
Hi, thanks. Can you please switch on the DEBUG logs and share them here?

**Comment by lukas-krecan** [comment]
Is it possible you have nested locks. That `task-that-fails` is calling other methods that in turn create a new lock. If that's the case, it explains the message. We have only one thread-local variable and as the thread is actually holding more locks, it behaves unexpectedly. Is it causing any issues or it's only about the WARN in the logs?

**Comment by m-mercuri** [comment]
Thanks for looking into this, @lukas-krecan. We don't have any nested locks in that process. I can put a breakpoint on the log line, what information can I look at when it stops there?

>  Is it causing any issues or it's only about the WARN in the logs?
For I could identify, it's not causing any issue, it's just the WARN.

**Comment by lukas-krecan** [comment]
Are you sure that `task-that-fails` is not triggering the other tasks? In the logs you can see that it obtains the lock, then the other tasks run, and then the `task-that-fails` finishes, all in the same thread. Moreover, it's the only way I was able to explain the behavior. 🤷‍♂️

**Comment by m-mercuri** [comment]
In the code path of the `task-that-fails` we don't have any part where we obtain another lock. It's true that the logs show them in the same thread, so I'm not sure how is that happening. Maybe the thread pool is reusing the thread. I'm gonna discuss it with my team and see if we can figure it out.

**Comment by m-mercuri** [comment]
One other thing, the logs that I provided are from my local dev application. As I mentioned I also saw it in our production logs, and in a couple of cases, I saw just right after the application started:
```
2023-07-24 11:29:58.364 INFO 1 --- [ main] io.company.Application : Starting Application using Java 17.0.7 on ...
....
2023-07-24 11:30:18.450 INFO 1 --- [ main] o.s.b.w.embedded.tomcat.TomcatWebServer : Tomcat started on port(s): 8080 (http) with context path ''
2023-07-24 11:30:18.726 INFO 1 --- [ main] io.company.Application : Started Application in 21.957 seconds (JVM running for 25.041)
2023-07-24 12:05:08.186 WARN 1 --- [   scheduling-5] n.j.s.core.DefaultLockingTaskExecutor    : No active lock, please report this as a bug.

```

**Comment by lukas-krecan** [comment]
Shoul be fixed in 5.7.0
